bounties_api/user/views.py

```diff
@@ -47,6 +51,46 @@ def get(self, request):
         return HttpResponse('Unauthorized', status=401)
 
 
+class RequestProfileImageUploadURL(APIView):
+    permission_classes = [AuthenticationPermission]
+
+    def get(self, request):
+        AWS_REGION = 'us-east-1'
+        client = boto3.client('s3', region_name=AWS_REGION)
+
+        current_user = request.current_user
+        nonce = int(random() * 1000)
+        bucket = 'assets.bounties.network'
+
+        sm_key = '{}/userimages/{}-sm-{}.png'.format(settings.ENVIRONMENT, current_user.public_address, nonce)
+        lg_key = '{}/userimages/{}-lg-{}.png'.format(settings.ENVIRONMENT, current_user.public_address, nonce)
+
+        sm_put_url = client.generate_presigned_url(
+            'put_object',
+            Params={
+                'ContentType': 'image/png',
+                'Bucket': bucket,
+                'ACL': 'public-read',
+                'Key': sm_key}
```

Comment:
Also - can you look up the thing I was telling you. There is a way to make it so the name of the file when you download it keeps its original name (without the nonce) although the key and url has the nonce.

Reply:
i also added this on the frontend!